02_activities/assignments/assignment_2.ipynb

```diff
@@ -1,339 +1,498 @@
+      "\n",
+      "0,1,1,3,3,4,4,6,3,4,9,9,7,6,8,15,12,15,6,11,6,18,5,14,15,12,9,8,3,6,10,6,8,7,2,5,4,3,1,1\n",
+      "\n",
+      "0,1,2,2,4,3,1,4,8,9,5,10,10,3,4,6,7,11,16,6,14,9,11,10,10,7,10,8,8,4,5,8,4,4,5,2,4,1,1,0\n",
+      "\n",
+      "0,0,2,3,4,5,4,6,2,9,7,4,9,10,8,11,16,12,15,17,19,10,18,13,15,11,8,4,7,11,6,7,6,5,1,3,1,0,0,0\n",
+      "\n",
+      "0,1,1,3,1,4,6,2,8,2,10,3,11,9,13,15,5,15,6,10,10,5,14,15,12,7,4,5,11,4,6,9,5,6,1,1,2,1,2,1\n",
+      "\n",
+      "0,0,1,3,2,5,1,2,7,6,6,3,12,9,4,14,4,6,12,9,12,7,11,7,16,8,13,6,7,6,10,7,6,3,1,5,4,3,0,0\n",
+      "\n",
+      "0,0,1,2,3,4,5,7,5,4,10,5,12,12,5,4,7,9,18,16,16,10,15,15,10,4,3,7,5,9,4,6,2,4,1,4,2,2,2,1\n",
+      "\n",
+      "0,1,2,1,1,3,5,3,6,3,10,10,11,10,13,10,13,6,6,14,5,4,5,5,9,4,12,7,7,4,7,9,3,3,6,3,4,1,2,0\n",
+      "\n",
+      "0,1,2,2,3,5,2,4,5,6,8,3,5,4,3,15,15,12,16,7,20,15,12,8,9,6,12,5,8,3,8,5,4,1,3,2,1,3,1,0\n",
+      "\n",
+      "0,0,0,2,4,4,5,3,3,3,10,4,4,4,14,11,15,13,10,14,11,17,9,11,11,7,10,12,10,10,10,8,7,5,2,2,4,1,2,1\n",
+      "\n",
+      "0,0,2,1,1,4,4,7,2,9,4,10,12,7,6,6,11,12,9,15,15,6,6,13,5,12,9,6,4,7,7,6,5,4,1,4,2,2,2,1\n",
+      "\n",
+      "0,1,2,1,1,4,5,4,4,5,9,7,10,3,13,13,8,9,17,16,16,15,12,13,5,12,10,9,11,9,4,5,5,2,2,5,1,0,0,1\n",
+      "\n",
+      "0,0,1,3,2,3,6,4,5,7,2,4,11,11,3,8,8,16,5,13,16,5,8,8,6,9,10,10,9,3,3,5,3,5,4,5,3,3,0,1\n",
+      "\n",
+      "0,1,1,2,2,5,1,7,4,2,5,5,4,6,6,4,16,11,14,16,14,14,8,17,4,14,13,7,6,3,7,7,5,6,3,4,2,2,1,1\n",
+      "\n",
+      "0,1,1,1,4,1,6,4,6,3,6,5,6,4,14,13,13,9,12,19,9,10,15,10,9,10,10,7,5,6,8,6,6,4,3,5,2,1,1,1\n",
+      "\n",
+      "0,0,0,1,4,5,6,3,8,7,9,10,8,6,5,12,15,5,10,5,8,13,18,17,14,9,13,4,10,11,10,8,8,6,5,5,2,0,2,0\n",
+      "\n",
+      "0,0,1,0,3,2,5,4,8,2,9,3,3,10,12,9,14,11,13,8,6,18,11,9,13,11,8,5,5,2,8,5,3,5,4,1,3,1,1,0\n",
+      "\n"
+     ]
+    }
+   ],
+   "source": [
+    "all_paths = [\n",
+    "  \"../../05_src/data/assignment_2_data/inflammation_01.csv\",\n",
+    "  \"../../05_src/data/assignment_2_data/inflammation_02.csv\",\n",
+    "  \"../../05_src/data/assignment_2_data/inflammation_03.csv\",\n",
+    "  \"../../05_src/data/assignment_2_data/inflammation_04.csv\",\n",
+    "  \"../../05_src/data/assignment_2_data/inflammation_05.csv\",\n",
+    "  \"../../05_src/data/assignment_2_data/inflammation_06.csv\",\n",
+    "  \"../../05_src/data/assignment_2_data/inflammation_07.csv\",\n",
+    "  \"../../05_src/data/assignment_2_data/inflammation_08.csv\",\n",
+    "  \"../../05_src/data/assignment_2_data/inflammation_09.csv\",\n",
+    "  \"../../05_src/data/assignment_2_data/inflammation_10.csv\",\n",
+    "  \"../../05_src/data/assignment_2_data/inflammation_11.csv\",\n",
+    "  \"../../05_src/data/assignment_2_data/inflammation_12.csv\"\n",
+    "]\n",
+    "with open(all_paths[0], 'r') as f:\n",
+    "    contents = f.readlines()\n",
+    "\n",
+    "for i in contents:\n",
```

Comment:
Question no. 1:
The for loop for 'contents' variable is not indented inside the 'with' block.

Although the contents variable is available outside of the 'with' block', contents is only relevant while the file is open and is tied to the file's content,. It would be more clear and maintainable to limit its scope to the 'with' block. Keeping variables like contents within the scope of the block makes the code more readable and shows that they are only valid as long as the file is open.